pkg/kubelet/cm/cpumanager/cpu_assignment.go

```diff
@@ -518,6 +575,50 @@ func (a *cpuAccumulator) takeFullSockets() {
 		a.take(cpusInSocket)
 	}
 }
+func (a *cpuAccumulator) takeFullUnCore() {
+	for _, uncore := range a.freeUncoreCache() {
+		cpusInUncore := a.topo.CPUDetails.CPUsInUncoreCaches(uncore)
+		if !a.needsAtLeast(cpusInUncore.Size()) {
+			continue
+		}
+		a.take(cpusInUncore)
+	}
+}
+
+// First try to take full UncoreCache, if available and need is at least the size of the UncoreCache group.
+// Second try to take the partial UncoreCache if available and the request size can fit w/in the UncoreCache.
+func (a *cpuAccumulator) takeUncoreCache() {
+	for _, uncore := range a.allUncoreCache() {
+		// take full UncoreCache if the CPUs needed is greater than free UncoreCache size
+		if a.numCPUsNeeded >= a.topo.NumCPUs/a.topo.NumUncoreCache {
+			a.takeFullUnCore()
+		}
+
+		if a.isSatisfied() {
+			return
+		}
+
+		// determine if SMT is enabled
+		numCoresNeeded := a.numCPUsNeeded / a.topo.CPUsPerCore()
+
+		var freeCPUsInUncoreCache cpuset.CPUSet
```

Comment:
ok, it took me a while to eventually realize this code implements what we agreed on the KEP, which in turn made me realize that we can use a simplification of the algorithm in the KEP, or at very least have a review of it and explain why we need to diverge from the flow here.
Since we are gathering quite a lot of todos, we may want to have another alpha cycle in 1.33. Let's see.

Reply:
Responded to first part in previous post.
No opposition to remain in alpha in 1.33 to iron more out.